storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,711 @@
+        String ret = "";
+        // we do this since to concat a null String will actually concat a "null", which is not the expected: ""
+        if (stormRoot != null) {
+            ret = stormRoot;
+        }
+        return ret + FILE_SEPARATOR + "stormdist";
+    }
+
+    public static String stormTmpPath(String stormRoot) {
+        return stormRoot + FILE_SEPARATOR + "tmp";
+    }
+
+    /* Never get used TODO : delete it*/
+    public static String masterTmpDir(Map conf) throws IOException {
+        String ret = stormTmpPath(masterLocalDir(conf));
+        FileUtils.forceMkdir(new File(ret));
+        return ret;
+    }
+
+    public static Map readSupervisorStormConfGivenPath(Map conf, String stormConfPath) throws  IOException {
+        Map ret = new HashMap(conf);
+        ret.putAll(Utils.fromCompressedJsonConf(FileUtils.readFileToByteArray(new File(stormConfPath))));
+        return ret;
+    }
+
+    /* Never get used TODO : may delete it*/
+    public static String masterStormMetaFilePath(String stormRoot) {
+        return (stormRoot + FILE_SEPARATOR + "storm-code-distributor.meta");
+    }
+
+    public static String masterStormJarPath(String stormRoot) {
+        return (stormRoot + FILE_SEPARATOR + "stormjar.jar");
+    }
+
+    /* Never get used TODO : may delete it*/
+    public static String masterStormCodePath(String stormRoot) {
+        return (stormRoot + FILE_SEPARATOR + "stormcode.ser");
+    }
+
+    /* Never get used TODO : may delete it*/
+    public static String masterStormConfPath(String stormRoot) {
+        return (stormRoot + FILE_SEPARATOR + "stormconf.ser");
+    }
+
+    public static String masterInbox(Map conf) throws IOException {
+        String ret = masterLocalDir(conf) + FILE_SEPARATOR + "inbox";
+        try {
+            FileUtils.forceMkdir(new File(ret));
+        } catch (IOException e) {
+            LOG.error("Failed to create dir " + ret, e);
+            throw e;
+        }
+        return ret;
+    }
+
+    public static String masterInimbusDir(Map conf) throws IOException {
+        return (masterLocalDir(conf) + FILE_SEPARATOR + "inimbus");
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedSupervisorLocalDir mocked = new SetMockedSupervisorLocalDir(dir)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedSupervisorLocalDir. dir)]
+    //     run test ...)
+    public static class SetMockedSupervisorLocalDir implements Closeable {
+        public SetMockedSupervisorLocalDir(String dir) {
+            mockedSupervisorLocalDir = dir;
+        }
+        @Override
+        public void close() {
+            mockedSupervisorLocalDir = null;
+        }
+    }
+    private static String mockedSupervisorLocalDir = null;
+    public static String supervisorLocalDir(Map conf) throws IOException {
+        if (mockedSupervisorLocalDir != null) {
+            return null;
```

Comment:
Then lets take the model and expand things to be a more complete mock, and we can do this everywhere.  I really would like consistency in how this works for all of the static mocking we do, and the real issue is we need to support returning a null value.

```
    public static class SetMockedSupervisorLocalDir implements Closeable {
        public SetMockedSupervisorLocalDir(String dir) {
            mockedSupervisorLocalDir = dir;
            isSupervisorLocalDirMocked = true;
        }
        @Override
        public void close() {
            isSupervisorLocalDirMocked = false;
            mockedSupervisorLocalDir = null;
        }
    }
    private static String mockedSupervisorLocalDir = null;
    private static boolean isSupervisorLocalDirMocked = false;

    public static String supervisorLocalDir(Map conf) throws IOException {
        if (isSupervisorLocalDirMocked) {
            return mockedSupervisorLocalDir;
    ...
```